conans/client/conan_api.py

```diff
@@ -644,29 +644,21 @@ def copy(self, reference, user_channel, force=False, packages=None):
                  self._user_io, self._remote_manager, self._registry, force=force)
 
     @api_method
-    def authenticate(self, name, password, remote=None):
-        if not remote:
-            remote = self._registry.default_remote
-        else:
-            remote = self._registry.remote(remote)
-        if password == "":
-            name, password = self._user_io.request_login(remote_name=remote.name,
-                                                         username=name)
-        self._remote_manager.authenticate(remote, name, password)
+    def authenticate(self, name, password, remote):
```

Comment:
We have to be consistent in the "remote" parameter in the whole API. I think (please check!) in all methods "remote" is a remote name. Here is a complex object. We could call `get_remote_by_name` inside, less efficient, but important to be consistent.

Reply:
Yes, please use "remote_name", when it is a string, use "remote" when it is a ``Remote`` object

Reply:
conan_api receives only remote names but param name is always "remote" instead fo "remote_name", so I sticked to that convention in this PR. I think it is better  to do that in a separated PR